Signal forms type checking fixes

crisbeto opened the pull request:
Includes the following fixes for type checking in signal forms:

### fix(compiler-cli): use any when checking field interface conformance
Switches to checking against `FormValueControl<any>` instead of `FormValueControl<unknown>` when checking whether custom controls conform to the interface.

### fix(compiler-cli): do not flag custom control required inputs as missing when field is present
Adds some logic that won't report the `value` or `checked` inputs as missing when the `Field` directive is present since it will bind to the inputs implicitly.

### fix(compiler-cli): allow value to be set on radio fields
Updates the logic that checks for unsupported bindigns to allow `value` to be set on `radio` controls.

### fix(compiler-cli): make field detection logic more robust
Currently the logic for detecting `Field` directives only works if it's imported from `@angular/forms/signals` which doesn't cover our own tests.

These changes make the check more robust.

### test(forms): fix type checking errors in signal form tests
Fixes some type checking errors in the signal forms tests that accumulated while there wasn't any type checking.

Fixes #64946.

JoostK reviewed (commented):
Could this have a test that shows the error isn't reported for the scenario this fixes?

crisbeto committed:
fix(compiler-cli): use any when checking field interface conformance

Switches to checking against `FormValueControl<any>` instead of `FormValueControl<unknown>` when checking whether custom controls conform to the interface.

Fixes #64946.

crisbeto committed:
fix(compiler-cli): do not flag custom control required inputs as missing when field is present

Adds some logic that won't report the `value` or `checked` inputs as missing when the `Field` directive is present since it will bind to the inputs implicitly.

crisbeto committed:
fix(compiler-cli): allow value to be set on radio fields

Updates the logic that checks for unsupported bindigns to allow `value` to be set on `radio` controls.

crisbeto committed:
fix(compiler-cli): make field detection logic more robust

Currently the logic for detecting `Field` directives only works if it's imported from `@angular/forms/signals` which doesn't cover our own tests.

These changes make the check more robust.

crisbeto committed:
test(forms): fix type checking errors in signal form tests

Fixes some type checking errors in the signal forms tests that accumulated while there wasn't any type checking.

leonsenft commented on packages/compiler-cli/src/ngtsc/typecheck/src/type_check_block.ts:
```diff
@@ -4124,6 +4112,27 @@ function hasModel(name: string, meta: TypeCheckableDirectiveMeta): boolean {
+  'readonly',
+  'touched',
+  'max',
+  'maxlength',
```
The lowercase variants (`maxlength`, `minlength`) that were included to catch unsupported attributes should not be included in this set of inputs. If a user's custom form control had a required `maxlength` input we wouldn't want to pretend it would be set by the `[field]` binding.

I think we can simply move them to the `unsupportedBindingFields` set.

crisbeto committed:
refactor(compiler-cli): update set of required inputs

Updates the required inputs that can be ignored for form controls.
